src/matrix.jl

```diff
@@ -95,6 +95,21 @@ function similar(M::IntervalMatrix)
     return IntervalMatrix(similar(M.mat))
 end
 
+# constructor from a scalar matrix
+function IntervalMatrix(A::AbstractMatrix{T}) where {T<:Number}
+    return IntervalMatrix(map(Interval, A))
+end
+
+# conversion from scalar matrix to interval matrix
+function convert(::Type{<:IntervalMatrix}, A::AbstractMatrix{T}) where {T<:Number}
+    return IntervalMatrix(map(x -> convert(Interval, x), A))
+end
```

Comment:
I would just use the first function in the second function.
```julia
function convert(::Type{<:IntervalMatrix}, A::AbstractMatrix{T}) where {T<:Number}
    return IntervalMatrix(A)
end
```

Reply:
But then this is not performing a conversion, right?

```julia
julia> x = Interval(1f0)
[1, 1]

julia> typeof(x)
Interval{Float32}


julia> convert(Interval{Float64}, x)
[1, 1]

julia> typeof(ans)
Interval{Float64}

Reply:
Then we wouldn't be doing any conversion; the intended behavior was to map the scalar case,

```julia
julia> x = Interval(1f0)
[1, 1]

julia> typeof(x)
Interval{Float32}

julia> convert(Interval{Float64}, x)
[1, 1]

julia> typeof(ans)
Interval{Float64}
```

Reply:
Okay, then there should be the faster `convert` for cases where the types agree.